aws/resources/ec2_egress_only_igw.go

```diff
@@ -0,0 +1,148 @@
+				Value: aws.String(util.FormatTimestamp(value)),
+			},
+		},
+	})
+	if err != nil {
+		return errors.WithStackTrace(err)
+	}
+
+	return nil
+}
+
+func (egigw *EgressOnlyInternetGateway) getFirstSeenTag(eoig ec2.EgressOnlyInternetGateway) (*time.Time, error) {
+	tags := eoig.Tags
+	for _, tag := range tags {
+		if util.IsFirstSeenTag(tag.Key) {
+			firstSeenTime, err := util.ParseTimestamp(tag.Value)
+			if err != nil {
+				return nil, errors.WithStackTrace(err)
+			}
+
+			return firstSeenTime, nil
+		}
+	}
+
+	return nil, nil
+}
+
+func shouldIncludeEgressOnlyInternetGateway(gateway *ec2.EgressOnlyInternetGateway, firstSeenTime *time.Time, configObj config.Config) bool {
+	var gatewayName string
+	// get the tags as map
+	tagMap := util.ConvertEC2TagsToMap(gateway.Tags)
+	if name, ok := tagMap["Name"]; ok {
+		gatewayName = name
+	}
+	return configObj.EgressOnlyInternetGateway.ShouldInclude(config.ResourceValue{
+		Name: &gatewayName,
+		Tags: tagMap,
+		Time: firstSeenTime,
+	})
+}
+
+func (egigw *EgressOnlyInternetGateway) getAll(_ context.Context, configObj config.Config) ([]*string, error) {
+	var result []*string
+
+	output, err := egigw.Client.DescribeEgressOnlyInternetGateways(&ec2.DescribeEgressOnlyInternetGatewaysInput{})
+	if err != nil {
+		return nil, errors.WithStackTrace(err)
+	}
+
+	for _, igw := range output.EgressOnlyInternetGateways {
+		// check first seen tag
+		firstSeenTime, err := egigw.getFirstSeenTag(*igw)
+		if err != nil {
+			logging.Errorf(
+				"Unable to retrieve tags for Egress IGW: %s, with error: %s", *igw.EgressOnlyInternetGatewayId, err)
+			continue
+		}
+
+		// if the first seen tag is not there, then create one
+		if firstSeenTime == nil {
+			now := time.Now().UTC()
+			firstSeenTime = &now
+			if err := egigw.setFirstSeenTag(*igw, time.Now().UTC()); err != nil {
+				logging.Errorf(
+					"Unable to apply first seen tag Egress IGW: %s, with error: %s", *igw.EgressOnlyInternetGatewayId, err)
+				continue
+			}
+		}
+
+		if shouldIncludeEgressOnlyInternetGateway(igw, firstSeenTime, configObj) {
+			result = append(result, igw.EgressOnlyInternetGatewayId)
+		}
+	}
+
+	// chgecking the nukable permissions
+	egigw.VerifyNukablePermissions(result, func(id *string) error {
+		_, err := egigw.Client.DeleteEgressOnlyInternetGateway(&ec2.DeleteEgressOnlyInternetGatewayInput{
+			EgressOnlyInternetGatewayId: id,
+			DryRun:                      aws.Bool(true),
+		})
+		return err
+	})
+
+	return result, nil
+}
+
+func (egigw *EgressOnlyInternetGateway) nukeAll(ids []*string) error {
+	if len(ids) == 0 {
+		logging.Debugf("No Egress only internet gateway ID's to nuke in region %s", egigw.Region)
+		return nil
+	}
+
+	logging.Debugf("Deleting all Egress only internet gateway in region %s", egigw.Region)
+	var deletedList []*string
+
+	for _, id := range ids {
+
+		if nukable, err := egigw.IsNukable(*id); !nukable {
+			logging.Debugf("[Skipping] %s nuke because %v", *id, err)
+			continue
```

Comment:
Is it fine that errors are ignored?

Reply:
Explain in the comment. Yeah it is safe to ignore in this cas.e